enos/modules/verify_secrets_engines/scripts/kv-pki-verify-certificates.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+set -e
+
+fail() {
+  echo "$1" 1>&2
+  exit 1
+}
+
+[[ -z "$MOUNT" ]] && fail "MOUNT env variable has not been set"
+[[ -z "$VAULT_ADDR" ]] && fail "VAULT_ADDR env variable has not been set"
+[[ -z "$VAULT_INSTALL_DIR" ]] && fail "VAULT_INSTALL_DIR env variable has not been set"
+[[ -z "$VAULT_TOKEN" ]] && fail "VAULT_TOKEN env variable has not been set"
+[[ -z "$COMMON_NAME" ]] && fail "COMMON_NAME env variable has not been set"
+[[ -z "$TTL" ]] && fail "TTL env variable has not been set"
+[[ -z "$TMP_TEST_RESULTS" ]] && fail "TMP_TEST_RESULTS env variable has not been set"
+
+binpath=${VAULT_INSTALL_DIR}/vault
+test -x "$binpath" || fail "unable to locate vault binary at $binpath" || fail "The certificate appears to be improperly configured or contains errors"
+export VAULT_FORMAT=json
+
+# Getting Certificates
+VAULT_CERTS=$("$binpath" list -format=json "${MOUNT}/certs" | jq -r '.[]')
+
+# Verify List Certificate
+[[ -z "$VAULT_CERTS" ]] && fail "VAULT_CERTS should include vault certificates"
+
+# Verifying Certificates
+for CERT in $VAULT_CERTS; do
+  echo "Getting Certificate from Vault PKI: ${CERT}"
+  "$binpath" read "${MOUNT}/cert/${CERT}" | jq -r '.data.certificate' > "${TMP_TEST_RESULTS}/tmp_vault_cert.pem"
+  echo "Verifying Certificate..."
+  openssl x509 -in "${TMP_TEST_RESULTS}/tmp_vault_cert.pem" -text -noout || fail "The certificate appears to be improperly configured or contains errors"
```

Comment:
This is good for checking the certificate formatting, but we probably want to verify the entire certificate chain.

Something like this will verify that the cert is valid:
`openssl verify -CAfile root_cert.pem -untrusted Intermediate.pem tmp_vault_cert.pem`

But openssl doesn't actually validate the whole chain with that. For that, you have to do each part individually.
To verify that the intermediate was actually issued by root you'll need to do something like:
`openssl verify -no-CAfile -no-CApath -partial_chain -trusted root_cert.pem Intermediate.pem`

And then verify that our actual cert was issued by our intermediate:
`openssl verify -no-CAfile -no-CApath -partial_chain -trusted Intermediate.pem tmp_vault_cert.pem`

Reply:
added steps to verify the valid cert